test/runtests.jl

```diff
@@ -168,6 +168,36 @@ end
     @test map(x -> x[2], Z) == B
     @test_throws BoundsError Z[-1]
 
+# Zip 2 named arrays.
+    Z = @inferred ZippedArray(A=A,B=B)
+    @test @inferred(ZippedArray((A=A,B=B))) === Z
+    @test @inferred(ZippedArray{Tuple{eltype(A),eltype(B)}}((A=A,B=B))) === Z
+    @test @inferred(ZippedArray{Tuple{eltype(A),eltype(B)}}(A=A,B=B)) === Z
+    @test @inferred(ZippedArray{Tuple{eltype(A),eltype(B)},ndims(A)}((A=A,B=B))) === Z
```

Comment:
Good if `eltype(A)` and `eltype(B)` are different (which I think is the case).

I have a concern about the order of the members: `ZippedArray(A=A,B=B)` and `ZippedArray(B=B,A=A)` are the same thing or a different one?